### DISCUSSION TITLE
Problem with annotations method

### DISCUSSION THREAD

**Discussion by Oskarovsky** [flapdoodle-oss/de.flapdoodle.embed.mongo#458] [Q&A] [answered]
I'm using the newest version of library --> de.flapdoodle.embed:de.flapdoodle.embed.mongo:4.6.2

When I'm trying to run tests I've got following errors:

```
/de/flapdoodle/embed/mongo/transitions/ImmutableMongodStarter.class): warning: Cannot find annotation method 'from()' in type 'Generated': class file for org.immutables.value.Generated not found
/de/flapdoodle/embed/mongo/transitions/ImmutableMongodStarter.class): warning: Cannot find annotation method 'generator()' in type 'Generated'
/de/flapdoodle/embed/mongo/transitions/ImmutableMongodStarter$Builder.class): warning: Cannot find annotation method 'from()' in type 'Generated'
/de/flapdoodle/embed/mongo/transitions/ImmutableMongodStarter$Builder.class): warning: Cannot find annotation method 'generator()' in type 'Generated'
/de/flapdoodle/reverse/transitions/ImmutableStart.class): warning: Cannot find annotation method 'from()' in type 'Generated'
/de/flapdoodle/reverse/transitions/ImmutableStart.class): warning: Cannot find annotation method 'generator()' in type 'Generated'
/de/flapdoodle/embed/mongo/transitions/ImmutableMongod.class): warning: Cannot find annotation method 'from()' in type 'Generated'
error: warnings found and -Werror specified
```

Do you know why it couldn'tfind this library -> org.immutables.value.Generated ?

/Oskar

**Comment by michaelmosmann** [comment]
@Oskarovsky the code is generated with https://immutables.github.io/ .. these annotations are part of the generated code but without any use.. i think you could get rid of these warnings if you add immutables.org as a dependency. Do you use spring with classpath scanning?

I will have a look if i can disable these annotations in the generated code.. just to get rid of these warnings..

**Reply by Oskarovsky** [reply]
Generally not. In places where I use your library, Spring has no specified places to scan

**Reply by Oskarovsky** [reply]
After adding the indicated dependency, Spring sees the org.immutables library, but it cannot find the Generated class (interface)

![image](https://user-images.githubusercontent.com/33788050/234490662-44bdf45c-f77c-47b2-bbed-737db8812590.png)

Of course, we can silence these errors by letting the app ignore the warnings, but I prefer not to do that in the app.

**Reply by michaelmosmann** [reply]
@Oskarovsky i agree .. i have created an issue for that: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/459

**Reply by michaelmosmann** [reply, marked as answer]
@Oskarovsky .. there is a new release (will take between 12 and 24hrs to be visible in maven central)

**Reply by Oskarovsky** [reply]
Thanks for quick fix ! :)

**Reply by michaelmosmann** [reply]
@Oskarovsky .. does it work?

**Reply by Oskarovsky** [reply]
@michaelmosmann 

I'm sorry you had to wait so long for an answer.
I used the new version 4.6.3 - the warning still occurs

.gradle\caches\modules-2\files-2.1\de.flapdoodle.reverse\de.flapdoodle.reverse\1.5.2\2e99b883bdeedbc47df87400db5d5df7be94ea4b\de.flapdoodle.reverse-1.5.2.jar(/de/flapdoodle/reverse/transitions/ImmutableStart.class): warning: Cannot find annotation method 'from()' in type 'Generated': class file for org.immutables.value.Generated not found

.gradle\caches\modules-2\files-2.1\de.flapdoodle.reverse\de.flapdoodle.reverse\1.5.2\2e99b883bdeedbc47df87400db5d5df7be94ea4b\de.flapdoodle.reverse-1.5.2.jar(/de/flapdoodle/reverse/transitions/ImmutableStart.class): warning: Cannot find annotation method 'generator()' in type 'Generated'

error: warnings found and -Werror specified

**Reply by michaelmosmann** [reply]
@Oskarovsky i will have a look at it

**Comment by michaelmosmann** [comment]
@Oskarovsky .. should be solved now.
